src/report_workflow/test_report_runner.py

```diff
@@ -207,48 +215,63 @@ def get_os_cluster_logs(base_path: str, test_number: str, test_type: str, compon
     return [os_stdout, os_stderr]
 
 
-def get_failed_tests(base_path: str, test_number: str, test_type: str, component_name: str, config: str,
-                     product_name: str) -> typing.List[list]:
-    failed_test: list = []
-    result_path: str = ''
-    result_content: str = ''
+def get_failed_tests(product_name: str, test_result: str, test_result_files: list) -> typing.List[list]:
```

Comment:
This method defaults to `.xml` and if OpenSearch will use the `html.parser` right ?

Reply:
See code:
```
        if self.name == "opensearch":
            soup = BeautifulSoup(result_content, "html.parser")
        else:
            soup = BeautifulSoup(result_content, "xml")

```